### PR TITLE
Refactor Token model to support modular blockchains and token types

### PR DESCRIPTION
**Pull request by Alexey1100**
1. Refactor Token model to extract blockchain-related data into Blockchain and TokenType classes.
2. Implement Blockchain and TokenType subclass template generators.

### CODE REVIEW

File: app/models/token.rb [hunk 1]
```diff
@@ -90,28 +85,98 @@ class Token < ApplicationRecord
...
   before_save :set_transitioned_to_ethereum_enabled
   before_save :enable_ethereum
   after_create :default_reg_group, if: -> { coin_type_comakery? }
 
+  def blockchain
+    "Blockchain::#{blockchain_network_to_subclass_name}".constantize.new
+  end
+
+  def blockchain_network_to_subclass_name
+    case blockchain_network
```
Comment (aquabu) [inline app/models/token.rb:113]: Just return the class constant?

File: app/decorators/account_decorator.rb [hunk 2]
```diff
@@ -53,10 +53,10 @@ def wallet_address_for(project)
 
   def wallet_address_url_for(project)
     address = wallet_address_for(project)
-    network = project.token&.blockchain_network
+    blockchain = project.token&.blockchain
 
-    if address.present? && network.present?
-      UtilitiesService.get_wallet_url(network, address)
+    if address.present? && blockchain.present?
+      blockchain.url_for_address_human(address)
```
Comment (aquabu) [inline app/decorators/account_decorator.rb:59]: maybe:
explorer_web_address_url
explorer_web_transaction_url
explorer_web_contract_url
explorer_web_host

File: app/javascript/components/TokenForm.jsx [hunk 3]
```diff
@@ -26,20 +26,22 @@ class TokenForm extends React.Component {
 
     this.lofoInputRef = React.createRef()
 
+    console.log(this.props)
```
Comment (aquabu) [inline app/javascript/components/TokenForm.jsx:29]: remove console.log

File: app/models/blockchain/bitcoin.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  # Is mainnet?
+  # @return [Boolean] mainnet?
+  def mainnet?
+    true
+  end
+
+  # Number of confirmations to wait before marking transaction as successful
+  # @return [Integer] number
+  def number_of_confirmations
+    1
```
Comment (aquabu) [inline app/models/blockchain/bitcoin.rb:32]: I think we may want to increase this to 3 or 4 confirmations.

**Comment by Alexey1100** [comment]
Noted.

At the moment we don’t have Bitcoin tx sync implemented.

**Review by aquabu — APPROVED** [review]

**Pull request merged**